src/Microsoft.ML.Data/StaticPipe/StaticSchemaShape.cs

```diff
@@ -105,15 +103,6 @@ public void Check(IExceptionContext ectx, SchemaShape shape)
                     // that the statically declared type should not have corresponded to a recognized type.
                     if (!pair.Value.IsAssignableFromStaticPipeline(type))
                     {
-                        // This is generally an error, unless it's the situation where the asserted type is Key<,> but we could
-                        // only resolve it so far as Key<>, since for the moment the SchemaShape cannot determine the type of key
-                        // value metadata. In which case, we can check if the declared type is a subtype of the key that was determined
-                        // from the analysis.
-                        if (pair.Value.IsGenericType && pair.Value.GetGenericTypeDefinition() == typeof(Key<,>) &&
-                                type.IsAssignableFromStaticPipeline(pair.Value))
-                        {
-                            continue;
-                        }
                         throw ectx.ExceptParam(nameof(shape),
                             $"Column '{pair.Key}' of type '{col.GetTypeString()}' cannot be expressed statically as type '{pair.Value}'.");
```

Comment:
>expressed statical [](start = 92, length = 18)

shall the message have a suggestion for how to fix it? Feels a bit cryptic to me